### ISSUE TITLE
Yet Another Issue Regarding xlwings and numpy

### ISSUE THREAD

**Issue by Portaloums** [xlwings/xlwings#1156]
OS = Windows 10
Versions = xlwings 0.15.10, Excel 2013, Python 3.6.9 (using Anaconda & Jupyter Notebooks)

I have been following along with several beginner xlwings Youtube tutorials, and now I'm following the examples pertaining to array formulas with numpy/pandas, as provided here: [https://docs.xlwings.org/en/stable/udfs.html](url)). I am encountering an issue with numpy when using the xlwings Excel add-in (my issue is very similar to the recent post from Rodrigo101032: [https://github.com/xlwings/xlwings/issues/1139](url)); however, I don't see where this issue was actually resolved. 

I am able to use numpy within a Jupyter notebook. There are no issues with importing numpy or, for example, performing basic matrix operations from within a Jupyter notebook.

Rather, after updating my UDF module, as shown below:

![numpyerror0](https://user-images.githubusercontent.com/48867998/66432871-10ce7e80-e9ed-11e9-811f-ae30def5566e.JPG)

...And then pressing the "Import Functions" button on the xlwings ribbon within Excel...

![numpyerror](https://user-images.githubusercontent.com/48867998/66432771-d5cc4b00-e9ec-11e9-97cd-d04b70e4dee5.JPG)

...I am thrown the following error message...

![numpyerror2](https://user-images.githubusercontent.com/48867998/66431155-77519d80-e9e9-11e9-9422-f48f1a99cd29.JPG)

This problem is occurring when I add "import numpy as np" and the matrix multiplication function (same function in Rodrigo101032's issue):

@xw.func
@xw.arg('x', np.array, ndim=2)
@xw.arg('y', np.array, ndim=2)
def matrix_mult(x, y):
    return x @ y

Without these lines of code, I am able to import my user defined functions into Excel.

Is it known whether or not this is an xlwings-specific issue? Granted, my error message does appear slightly different from those described in otherwise similar issues, so perhaps it is something else entirely (??).

For full disclosure, I have attempted uninstalling and reinstalling numpy (version 1.16.5).  I have also been used [https://pythonforundergradengineers.com/new-virtual-environment-with-conda.html](url) as a guide on creating virtual environments with Conda, and I have downloaded packages, such as xlwings and numpy, within a virtual environment dedicated to tinkering with xlwings.

Any guidance is very appreciated!

**Comment by fzumstein** [comment]
Can you please provide your full config, including the Conda part?

**Comment by Portaloums** [comment]
@fzumstein Yes, absolutely, although I'm not sure what exactly is meant by "full config". What other information would be helpful? 

Earlier, I created an environment called "xlwings_tutor" via Anaconda Prompt. Then, within this environment, I installed xlwings and numpy using conda install.

![anaconda_prompt](https://user-images.githubusercontent.com/48867998/66448544-d9c59080-ea1f-11e9-9021-066c28d972a4.JPG)

**Comment by fzumstein** [comment]
I meant your settings that you have in the ribbon

**Comment by Portaloums** [comment]
Sure thing. Here's the entire ribbon:

![xlwings_ribbon](https://user-images.githubusercontent.com/48867998/66483004-2fc32400-ea72-11e9-8fe3-1b608bcb708e.JPG)

**Comment by fzumstein** [comment]
You will need to provide the conda configuration: http://docs.xlwings.org/en/stable/addin.html#global-settings

Please use a conda env, as the base environment might still be buggy (there's an open issue for that)

**Comment by Portaloums** [comment]
@fzumstein Ah, I see...OK, so I've had some success. I am now able to call the numpy-dependent matrix multiplication function from within Excel. For others with similar issues in the future (and, Felix, for your reference), I took the following steps:

(0) Previously, I had not entered anything into the Conda Base or Conda Env fields (found in the ribbon, shown above). 

(1) I added the name of my Conda environment into the corresponding field in the ribbon, and then pressed the Import Functions button. I received the same error as before.

(2) After reading the link Felix provided above, I noticed that in many cases it is required to also provide the full path of the Conda Base. At first, I accidentally provided the full path to the environments folder, which yielded the following different error message:

![MVB_error1](https://user-images.githubusercontent.com/48867998/66493222-acaac980-ea83-11e9-8ca9-5c165e715f04.JPG)

(3) Then, I realized my mistake and that I had to shorten the path to just the base environment. Upon doing this and then clicking the Import Functions button, no error message was thrown and an instance of my conda environment opens in a new cmd prompt window, shown here:

![whathappened1](https://user-images.githubusercontent.com/48867998/66493466-1dea7c80-ea84-11e9-9219-7b779cb7c278.JPG)

And now it looks like I'm not having any issues with numpy via xlwings add-in:

![mm](https://user-images.githubusercontent.com/48867998/66493645-66a23580-ea84-11e9-990e-6686c0deaeac.JPG)

I hope this is helpful to others wishing to import custom user-defined functions from Python. Again, @fzumstein, your assistance is greatly appreciated (and xlwings is extremely useful).
